examples/pytorch_to_stockroom_loader.py

```diff
@@ -0,0 +1,130 @@
+    columns_added = []
+    for i, column in enumerate(sample):
+
+        if isinstance(column, int):
+            prototype_int = np.array([column])
+            column_name = f'{name}_{column_postfixes[i]}'
+            columns_added.append(column_name)
+            master.add_ndarray_column(
+                column_name,
+                prototype=prototype_int)
+            print(f'Created {column_name}')
+
+        elif isinstance(column, torch.Tensor):
+            numpy_col = column.detach().numpy()
+            column_name = f'{name}_{column_postfixes[i]}'
+            columns_added.append(column_name)
+            master.add_ndarray_column(
+                column_name,
+                prototype=numpy_col)
+            print(f'Created {column_name}')
+
+        elif isinstance(column, list):
+            # nested columns
+            if isinstance(column[0], str):
+                column_name = f'{name}_{column_postfixes[i]}'
+                columns_added.append(column_name)
+                master.add_str_column(column_name,
+                                      contains_subsamples=True)
+                print(f'Created subsampled column {column_name}')
+
+    # commit and close hangar repo
+    master.commit('added columns'+str(columns_added))
+    master.close()
+
+    assert len(sample) == len(columns_added), f'{len(sample)} != {len(columns_added)}'
+    # open stockroom
+    stock = StockRoom(root_dir, write=True)
+
+    with stock:
+        with stock.run():
+            for i, sample in tqdm(enumerate(dataset), total=len(dataset),
+                                  position=0, leave=True):
+                for column, item in zip(columns_added, sample):
+                    stock.data[column][i] = hangar_transform(item)
+
+def main(args):
+    print(args)
+
+    # build the config for the Dataset
+    dataset_config = dataset_configs[args.dataset]
+    d_names, kwargs = dataset_config.gen_kwargs(args.root)
+    print(kwargs)
+
+    datasets_configured = []
+    for i, name in enumerate(d_names):
+        d = datasets.__dict__[dataset_config.name](**kwargs[i])
+        datasets_configured.append(d)
+        print(d)
+
+    # add to stockroom
+    for i, d in enumerate(datasets_configured):
+        add_from_dataset(d_names[i], d, args.root, ['img', 'label'])
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('dataset',
+                        type=str,
+                        choices=list(dataset_configs.keys()),
+                        help='The name of PyTorch dataset you want to add to stockroom')
+
+    parser.add_argument('--root',
+                        type=str,
+                        help='Root directory in which the dataset is downloaded to',
+                        default='./')
+    parser.add_argument('--download', '-d',
+                        action='store_true',
+                        default=False)
+
+    args = parser.parse_args()
+    main(args)
```

Comment:
@hhsecond let me know how you feel about using `click` instead of `Argparse` here.. I find it so much more intuitive / readable / performant, but I'm not sure which direction you want to take stockroom in.

Reply:
I agree @rlizzo. The current CLI is stockroom is built on `click`.
Probably information for later; I am also exploring [type](https://typer.tiangolo.com/) which is a nice wrapper over click. In any case, I am not planning to change that for this release

@jjmachan, would you mind migrating the PR to click?

Reply:
on it :wink: